src/Plugin/TfaValidation/TfaHotp.php

```diff
@@ -0,0 +1,249 @@
+<?php
+
+/**
+ * @file TfaHOTP class
+ */
+
+namespace Drupal\tfa\Plugin\TfaValidation;
+
+use Base32\Base32;
+use Drupal\Core\Extension\ModuleHandlerInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Site\Settings;
+use Drupal\user\UserDataInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\tfa\Plugin\TfaBasePlugin;
+use Drupal\tfa\Plugin\TfaValidationInterface;
+use Otp\GoogleAuthenticator;
+use Otp\Otp;
+use Drupal\Core\DependencyInjection\DependencySerializationTrait;
+
+/**
+ * @TfaValidation(
+ *   id = "tfa_hotp",
+ *   label = @Translation("TFA Hotp"),
+ *   description = @Translation("TFA Hotp Validation Plugin"),
+ *   fallbacks = {
+ *    "tfa_recovery_code"
+ *   }
+ * )
+ */
+class TfaHotp extends TfaBasePlugin implements TfaValidationInterface {
+  use DependencySerializationTrait;
+
+  /**
+   * @var GoogleAuthenticator
+   */
+  protected $auth;
+
+  /**
+   * Provides the user data service object.
+   *
+   * @var \Drupal\user\UserDataInterface
+   */
+  protected $userData;
+
+  /**
+   * @var int
+   */
+  protected $timeSkew;
+
+  /**
+   * @var bool
+   */
+  protected $alreadyAccepted;
+
+  /**
+   * @copydoc TfaBasePlugin::__construct()
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, UserDataInterface $user_data) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->auth = new \StdClass();
+    $this->auth->otp = new Otp();
+    $this->auth->ga  = new GoogleAuthenticator();
```

Comment:
Yes each individual validation has an id and more metadata which is defined in these classes.
Also every validation has a different validation methods, the steps differ like HOTP has the second factor which needs to be handled differently than the one in TOTP